language-server/src/main/java/org/sonarlint/languageserver/SonarLintLanguageServer.java

```diff
@@ -582,6 +581,21 @@ private static URI parseURI(String uriStr) {
     return uri;
   }
 
+  @Override
+  public CompletableFuture<Map<String, List<RuleDescription>>> listAllRules() {
+    Map<String, List<RuleDescription>> result = new HashMap<>();
+    Map<String, String> languagesNameByKey = engineCache.getOrCreateStandaloneEngine().getAllLanguagesNameByKey();
+    engineCache.getOrCreateStandaloneEngine().getAllRuleDetails()
+      .forEach(d -> {
+        String languageName = languagesNameByKey.get(d.getLanguageKey());
+        if (!result.containsKey(languageName)) {
+          result.put(languageName, new ArrayList<>());
+        }
+        result.get(languageName).add(RuleDescription.of(d));
+      });
+    return CompletableFuture.completedFuture(result);
```

Comment:
Change to use the correct pattern for cancellation support. See my other PR for examples.